Todos and rxjs

VeronikaSoshenko opened the pull request:
Сделала юзерам сервис и повторила все для задач

VeronikaSoshenko committed:
verstka

VeronikaSoshenko committed:
Update app.component.html

Удалили поля

VeronikaSoshenko committed:
js

Поправила макет: хэдер корзина, все карточки съехавшие выстроила по размерам, стрелки поставила на место.

VeronikaSoshenko committed:
Дз Js

Исправила в коде мелкие ошибки, выявленные в прошлом код ревью. Перенесла стили прописанные в html в css. Выполнила домашние задания 1-3, 4 тоже, но не понимаю почему оно не работает

VeronikaSoshenko committed:
Роутинг, создание компонента, запрос на сервер

Разделила на отдельные компоненты хидер(новая папка) и мейн боди(оставила в основной папке)

VeronikaSoshenko committed:
resolve-conflicts, add header

VeronikaSoshenko committed:
Merge branch 'routing-2' into routing

VeronikaSoshenko committed:
Merge pull request #1 from Safarov12/routing

Слили ветку с кодом в ветку с картинками

VeronikaSoshenko committed:
Роутинг, создание компонента, запрос на сервер

Сделала задачи по JS, но нужна помощь с 3,8,9. (дз по адресу /hw ) Поработала с юзерами и компонентами

VeronikaSoshenko committed:
Сервисы, классы, передача данных в компонент

Сделала функционал с задачами

VeronikaSoshenko committed:
RXJS

Сделала юзерам сервис и повторила все для задач

pocahontassss commented on src/app/user-list/user-list.component.html:
```diff
@@ -1,5 +1,5 @@
 <div class="content-container_users">
-  <div *ngFor="let item of users">
+  <div *ngFor="let item of usersService.userSubject$ | async">
```
Ты создала поток данных userSubject. После ты должна была сделать его неизменяемым, то-есть применить оператор .asObservable, и потом с этим новым потоком работать, а так ты обращаешься к сабджекту.

pocahontassss commented on src/app/todos-list/todos-card/todos-card.component.ts:
```diff
@@ -8,7 +8,6 @@ import { Component, EventEmitter, Input, Output } from "@angular/core";
-
 export class TodosCardComponent {
   @Input () 
   todo: any
```
Нужно протипизировать пришедшие данные через инпут

pocahontassss commented on src/app/todos-list/todos-list.component.html:
```diff
@@ -6,7 +6,7 @@ <h2>
 
 <div class="content-container_todos">
-  <div *ngFor="let todo of todos">
+  <div *ngFor="let todo of todosService.todoSubject$ | async">
```
Аналогично, как и с пользователями, нужно обращаться к обсерваблу

pocahontassss commented on src/app/todo.service.ts:
```diff
@@ -0,0 +1,47 @@
+    )
+  }
+
+  deleteTodo (id: any) {
```
Нужно протипизировать

RYAZHAPOVILNUR commented:
я тоже проверил, согласен с комментами Владислава

VeronikaSoshenko committed:
users and todos form

Исправила ошибки с прошлого дз ( добавила asObservable и протипизировала данные пришедшие через input). Создала формы для юзеров и задачекю

pocahontassss commented on src/app/user-list/user-list.component.ts:
```diff
@@ -44,20 +41,27 @@ export interface User {
   deleteUser (id: number) {
     this.usersService.deleteUser(id)
   }
+
+  public createUser (formData: any) {
+    this.usersService.createUser({
```
Нужно протипизировать

pocahontassss commented on src/app/todos-list/todos-list.component.ts:
```diff
@@ -33,4 +35,13 @@ export class TodosListComponent {
   DeleteTodo(id:number) {
     this.todosService.deleteTodo(id)
   }
+
+  public CreateTodo (formData:any) {
+    this.todosService.createTodo({
```
Нужно протипизировать

pocahontassss replied:
Нужна типизация

pocahontassss commented on src/app/user-list/user-list.component.ts:
```diff
@@ -44,20 +41,27 @@ export interface User {
   deleteUser (id: number) {
     this.usersService.deleteUser(id)
   }
+
+  public createUser (formData: any) {
+    this.usersService.createUser({
+      id: new Date().getTime(),
```
Нужна типизация

VeronikaSoshenko committed:
Todo form fix

Исправила баг с формой todo для поля выполнения задач. Протипизировала данные из замечения к прошлому дз

pocahontassss reviewed (changes requested):
Задачи протипизировала, а юзеров?)

VeronikaSoshenko committed:
Убрала any для юзеров

Протипизировала юзеров, создав экспорт интерфейса

VeronikaSoshenko committed:
Вынесла интерфейсы в отдельную папку

Создала папку интерфейсы, в ней два файла с интефейсами для задач и юзеров
